impl/src/main/java/org/ehcache/internal/store/tiering/CacheStore.java

```diff
@@ -188,10 +194,34 @@ public boolean replace(K key, V oldValue, V newValue) throws CacheAccessExceptio
 
   @Override
   public void clear() throws CacheAccessException {
+    boolean interrupted = false;
+    while(!cachingTierRef.compareAndSet(realCachingTier, noopCachingTier)) {
+      synchronized (noopCachingTier) {
+        if(cachingTierRef.get() == noopCachingTier) {
+          try {
+            noopCachingTier.wait();
+          } catch (InterruptedException e) {
+            interrupted = true;
+          }
+        }
+      }
+    }
     try {
       authoritativeTier.clear();
     } finally {
-      cachingTier.invalidate();
+      try {
+        realCachingTier.invalidate();
+      } finally {
+        if(interrupted) {
```

Comment:
This probably should be moved to before the authoritative clear call. We don't know whether a third party's authoritative tier implementation might be using interrupts.